### PR TITLE
option to pass dataloader directly (rather than passing dataset)

### PR DESCRIPTION
**Pull request by oliveradk**

### CODE REVIEW

File: src/cupbearer/detectors/anomaly_detector.py [hunk 1]
```diff
@@ -89,7 +89,8 @@ def _finetune(self, **kwargs) -> dict:
 
     def eval(
         self,
-        dataset: MixedData,
+        dataset: MixedData | None,
```
Comment (ejnnr) [inline src/cupbearer/detectors/anomaly_detector.py:92]: Is not having a default value for this deliberate? Seems a bit cumbersome if used with a dataloader. If we make both optional, then we should probably have a meaningful error message for when neither is provided.

**Review by ejnnr — APPROVED** [review]
Looks good, I don't have a strong take on whether `dataset` should be optional. Curious what the use case is, do you just want to customize `num_workers` or other things? We could just make all of those arguments to `eval()`, but that might get a bit much. Happy for you to merge this (or make any changes and then merge).

File: src/cupbearer/detectors/anomaly_detector.py (hunk 1 above)
Comment (oliveradk) [reply src/cupbearer/detectors/anomaly_detector.py]: Nah I agree, adding a default argument seems better

**Comment by oliveradk** [comment]
(usecase is using a custom DataLoader class with a particular collate_fn function, amoung other things (for working with the auto_circuits library))

**Pull request merged**